test/utils/cartesian_hashtable_test.py

```diff
@@ -0,0 +1,296 @@
+__author__ = "Riccardo De Zen <riccardodezen98@gmail.com>"
+__version__ = "1.0"
+
+from otri.utils.cartesian_hashtable import CartesianHashTable
+
+from typing import Iterable, Any, Callable
+from numbers import Real
+import numpy
+import unittest
+from parameterized import parameterized_class, parameterized
+
+
+def origin(item: Any):
+    '''
+    Return two axis cartesian origin.
+    '''
+    return (0, 0)
+
+
+def cartesian_tuple(item: Iterable[Real]):
+    '''
+    Return the item itself converted to a tuple.
+    '''
+    return tuple(item)
+
+
+def are_near(item: Iterable[Real], other: Iterable[Real], approx: Real, coords: Callable) -> bool:
+    '''
+    Return True if the coordinates of the two items are near enough.
+
+    Parameters:
+        item : Iterable[Real]
+            Item's coordinates are used as the 100%.
+    '''
+    approx = abs(approx)
+    item_coords = coords(item)
+    other_coords = coords(other)
+    for i in range(len(item_coords)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=6277039)